integrations/outlook/src/index.ts

```diff
@@ -0,0 +1,252 @@
+     */
+  },
+  actions: {},
+  channels: {
+    channel: {
+      messages: {
+        text: async (props) => {
+          const graphClient = getClient(props.ctx.configuration)
+          await graphClient.sendMail({
+            ...props,
+            body: {
+              contentType: 'text',
+              content: `${props.payload.text}\n\n${props.ctx.configuration.emailSignature}`,
+            },
+          })
+        },
+        image: async () => {
+          throw new NotImplementedError()
+        },
+        markdown: async () => {
+          throw new NotImplementedError()
+        },
+        audio: async () => {
+          throw new NotImplementedError()
+        },
+        video: async () => {
+          throw new NotImplementedError()
+        },
+        file: async () => {
+          throw new NotImplementedError()
+        },
+        location: async () => {
+          throw new NotImplementedError()
+        },
+        carousel: async () => {
+          throw new NotImplementedError()
+        },
+        card: async () => {
+          throw new NotImplementedError()
+        },
+        choice: async (props) => {
+          const graphClient = getClient(props.ctx.configuration)
+          let content = `${props.payload.text}\n`
+
+          for (const option of props.payload.options) {
+            content += `- ${option.label}\n`
+          }
+          content += `\n\n${props.ctx.configuration.emailSignature}`
+          await graphClient.sendMail({
+            ...props,
+            body: {
+              contentType: 'text',
+              content,
+            },
+          })
+        },
+        dropdown: async (props) => {
+          const graphClient = getClient(props.ctx.configuration)
+          let content = `${props.payload.text}\n`
+
+          for (const option of props.payload.options) {
+            content += `- ${option.label}\n`
+          }
+          content += `\n\n${props.ctx.configuration.emailSignature}`
+          await graphClient.sendMail({
+            ...props,
+            body: {
+              contentType: 'text',
+              content,
+            },
+          })
+        },
+        html: async (props) => {
+          const graphClient = getClient(props.ctx.configuration)
+          await graphClient.sendMail({
+            ...props,
+            body: {
+              contentType: 'html',
+              content: props.payload.content,
+            },
+          })
+        },
+      },
+    },
+  },
+  handler: async ({ req, ctx, client }) => {
+    // If there is a validationToken parameter
+    // in the query string, this is the endpoint validation
+    // request sent by Microsoft Graph. Return the token
+    // as plain text with a 200 response
+    // https://docs.microsoft.com/graph/webhooks#notification-endpoint-validation
+    if (req.query) {
+      const queryString = req.query as string
```

Comment:
**chore:** there are node librairies to parse query string. This obfuscate the behavior of the function.